src/main/java/io/cdap/plugin/gcp/dataplex/source/DataplexBatchSource.java

```diff
@@ -259,7 +261,17 @@ private void configureBigQuerySource() {
 
   private void emitLineage(BatchSourceContext context, Schema schema, TableDefinition.Type sourceTableType,
                            String table) {
-    LineageRecorder lineageRecorder = new LineageRecorder(context, config.getReferenceName());
+    // entity.getDataPath() will be of format:
+    // 'projects/datasetProjectName/datasets/datasetName/tables/tableName'
+    String[] dataPathValues = entity.getDataPath().split("/");
```

Comment:
I see there is a getEntityValuesFromDataPathForBQEntities() method that sets the dataset and table variables. Can that be re-used or is there some reason that we need to parse this again? I'm also not clear why table is passed in as an argument even though it is in the entity, is it possible for the table that is passed into this method to ever differ from what is parsed from the entity?

FYI, the existing code isn't structured very well, we shouldn't be using shared state like this, as it's not clear when everything gets assigned a value. It would have been much better to parse it once into some id class that is composed of a project/dataset/table and pass that around everywhere. You don't need to do any refactoring around this in the PR, but just pointing out that this is a pattern to avoid.

Reply:
Ack. Must've missed `getEntityValuesFromDataPathForBQEntities()`.